src/app/color-map-menu.component.ts

```diff
@@ -24,7 +24,61 @@ export class ColorMapMenuComponent {
     }>;
   }>;
 
-  constructor(@Inject(MAT_DIALOG_DATA) public data: any) {
+  min = 0;
+  max = 17;
+  inverse = false;
+
+  // closes color map menu
+  closeColorMaps() {
+    this.dialogRef.close();
+  }
+
+  // checks if the current index is the inverse version. Displays depending on the state of the inverse switch
+  indexSkip(num) {
+    if (this.inverse) {
+      if (num % 2 === 1) {
+        return true;
+      }
+    } else {
+      if (num % 2 === 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // inverse switch
+  toggleInverseMaps() {
+    this.inverse ? (this.inverse = false) : (this.inverse = true);
+  }
+
+  // method to change the range of the ColorMap for loop
+  changeColors(btn) {
```

Comment:
Here, in the function, add a dictionary (key, value pair lookup), where you pass a string name of what you want, and the pair would be another dictionary with the "min" and "max" values.

Another upgrade (not needed for this PR, but something to think about), is to update my crappy original colorMaps.json file and don't make it a list of colormaps, but make it a key/value pair of color maps with the Key the name of the colormap. You don't need to do this for this PR, just another update we could do.